From 052fe1977eb0618a0b8afda57082c3de81b31ff8 Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Thu, 2 Jan 2020 17:39:36 +0800
Subject: [PATCH] system/locallogin.te: allow login read symlink file

fix avc denials:

AVC avc:  denied  { read } for  pid=512 comm="login" name="root"
dev="sda3" ino=869 scontext=system_u:system_r:local_login_t:s0-s15:c0.c1023
tcontext=system_u:object_r:default_t:s0 tclass=lnk_file permissive=0

Upstream-Status: Pending

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 policy/modules/system/locallogin.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/locallogin.te b/policy/modules/system/locallogin.te
index f629b0040..8d80e877b 100644
--- a/policy/modules/system/locallogin.te
+++ b/policy/modules/system/locallogin.te
@@ -103,6 +103,8 @@ files_read_world_readable_sockets(local_login_t)
 # for when /var/mail is a symlink
 files_read_var_symlinks(local_login_t)
 
+files_read_all_symlinks(local_login_t)
+
 fs_search_auto_mountpoints(local_login_t)
 
 storage_dontaudit_getattr_fixed_disk_dev(local_login_t)
-- 
2.17.1

